base/inference.jl

```diff
@@ -3245,11 +3257,12 @@ end
 
 #### finalize and record the result of running type inference ####
 
-function isinlineable(m::Method, src::CodeInfo)
+function isinlineable(m::Method, src::CodeInfo, mod::Module, params::InferenceParams, bonus::Int=0)
     # compute the cost (size) of inlining this code
     inlineable = false
-    cost = 1000
+    cost_threshold = params.inline_cost_threshold
     if m.module === _topmod(m.module)
+        # a few functions get special treatment
```

Comment:
Perhaps `throw` should be treated specially as well, to avoid the necessity of wrapping `throw`s into `@noinline`'d methods?